### PR TITLE
refactor(lane_change): separate lane change and external request

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4e07f4e</samp>

This pull request refactors the lane change functionality of the behavior path planner into two separate packages: `behavior_path_lane_change_module` and `behavior_path_external_request_lane_change_module`. The former handles the normal lane change behavior, while the latter handles the lane change requests from external sources. The pull request also updates the include paths, the package dependencies, the documentation, and the tests to reflect the new structure.

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_lane_change_module/test/test_behavior_path_planner_node_interface.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  test_utils::updateNodeOptions(
+    node_options,
+    {planning_test_utils_dir + "/config/test_common.param.yaml",
+     planning_test_utils_dir + "/config/test_nearest_search.param.yaml",
+     planning_test_utils_dir + "/config/test_vehicle_info.param.yaml",
+     behavior_path_planner_dir + "/config/behavior_path_planner.param.yaml",
+     behavior_path_planner_dir + "/config/drivable_area_expansion.param.yaml",
+     behavior_path_planner_dir + "/config/scene_module_manager.param.yaml",
+     behavior_path_planner_dir + "/config/dynamic_avoidance/dynamic_avoidance.param.yaml",
```
Comment (satoshi-ota) [inline planning/behavior_path_lane_change_module/test/test_behavior_path_planner_node_interface.cpp:67]: @zulfaqar-azmi-t4 Unnecessary?

File: planning/behavior_path_planner/test/test_behavior_path_planner_node_interface.cpp [hunk 2]
```diff
@@ -49,7 +49,6 @@ std::shared_ptr<BehaviorPathPlannerNode> generateNode()
     ament_index_cpp::get_package_share_directory("behavior_path_planner");
 
   std::vector<std::string> module_names;
-  module_names.emplace_back("behavior_path_planner::DynamicAvoidanceModuleManager");
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/test/test_behavior_path_planner_node_interface.cpp:52]: @zulfaqar-azmi-t4 I guess you don't have to remove this line.

File: planning/behavior_path_external_request_lane_change_module/test/test_behavior_path_planner_node_interface.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  test_utils::updateNodeOptions(
+    node_options,
+    {planning_test_utils_dir + "/config/test_common.param.yaml",
+     planning_test_utils_dir + "/config/test_nearest_search.param.yaml",
+     planning_test_utils_dir + "/config/test_vehicle_info.param.yaml",
+     behavior_path_planner_dir + "/config/behavior_path_planner.param.yaml",
+     behavior_path_planner_dir + "/config/drivable_area_expansion.param.yaml",
+     behavior_path_planner_dir + "/config/scene_module_manager.param.yaml",
+     behavior_path_planner_dir + "/config/dynamic_avoidance/dynamic_avoidance.param.yaml",
```
Comment (satoshi-ota) [inline planning/behavior_path_external_request_lane_change_module/test/test_behavior_path_planner_node_interface.cpp:67]: @zulfaqar-azmi-t4 Can remove...?

**Comment by satoshi-ota** [comment]
@zulfaqar-azmi-t4 Basically, it looks good to me. After fixing above suggestions, I'll approve this PR.

File: planning/behavior_path_external_request_lane_change_module/test/test_behavior_path_planner_node_interface.cpp (hunk 3 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_external_request_lane_change_module/test/test_behavior_path_planner_node_interface.cpp]: fixed in [010c69e](https://github.com/autowarefoundation/autoware.universe/pull/5850/commits/010c69ee8748019a15eefd08e47595a9b90ce8d8)

File: planning/behavior_path_planner/test/test_behavior_path_planner_node_interface.cpp (hunk 2 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/test/test_behavior_path_planner_node_interface.cpp:52]: [010c69e](https://github.com/autowarefoundation/autoware.universe/pull/5850/commits/010c69ee8748019a15eefd08e47595a9b90ce8d8)

File: planning/behavior_path_lane_change_module/test/test_behavior_path_planner_node_interface.cpp (hunk 1 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_lane_change_module/test/test_behavior_path_planner_node_interface.cpp]: fixed in [010c69e](https://github.com/autowarefoundation/autoware.universe/pull/5850/commits/010c69ee8748019a15eefd08e47595a9b90ce8d8)

**Comment by satoshi-ota** [comment]
@zulfaqar-azmi-t4 colcon test seems to fail... :thinking:

File: planning/behavior_path_external_request_lane_change_module/test/test_behavior_path_planner_node_interface.cpp [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    node_options,
+    {
+      planning_test_utils_dir + "/config/test_common.param.yaml",
+      planning_test_utils_dir + "/config/test_nearest_search.param.yaml",
+      planning_test_utils_dir + "/config/test_vehicle_info.param.yaml",
+      behavior_path_planner_dir + "/config/behavior_path_planner.param.yaml",
+      behavior_path_planner_dir + "/config/drivable_area_expansion.param.yaml",
+      behavior_path_planner_dir + "/config/scene_module_manager.param.yaml",
+      ament_index_cpp::get_package_share_directory("behavior_path_lane_change_module") +
+        "/config/lane_change/lane_change.param.yaml",
```
Comment (satoshi-ota) [inline planning/behavior_path_external_request_lane_change_module/test/test_behavior_path_planner_node_interface.cpp:69]:
```suggestion
        "/config/lane_change.param.yaml",
```

File: planning/behavior_path_lane_change_module/test/test_behavior_path_planner_node_interface.cpp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  node_options.parameter_overrides(params);
+
+  test_utils::updateNodeOptions(
+    node_options, {planning_test_utils_dir + "/config/test_common.param.yaml",
+                   planning_test_utils_dir + "/config/test_nearest_search.param.yaml",
+                   planning_test_utils_dir + "/config/test_vehicle_info.param.yaml",
+                   behavior_path_planner_dir + "/config/behavior_path_planner.param.yaml",
+                   behavior_path_planner_dir + "/config/drivable_area_expansion.param.yaml",
+                   behavior_path_planner_dir + "/config/scene_module_manager.param.yaml",
+                   behavior_path_planner_dir + "/config/lane_change/lane_change.param.yaml"});
```
Comment (satoshi-ota) [inline planning/behavior_path_lane_change_module/test/test_behavior_path_planner_node_interface.cpp:66]:
```suggestion
                   behavior_path_planner_dir + "/config/lane_change.param.yaml"});
```

**Comment by satoshi-ota** [comment]
@zulfaqar-azmi-t4 I missed AbLC test.
```
1: -- run_test.py: invoking following command in '/__w/autoware.universe/autoware.universe/build/behavior_path_avoidance_by_lane_change_module':
1:  - /__w/autoware.universe/autoware.universe/build/behavior_path_avoidance_by_lane_change_module/test_behavior_path_avoidance_by_lane_change_module --gtest_output=xml:/__w/autoware.universe/autoware.universe/build/behavior_path_avoidance_by_lane_change_module/test_results/behavior_path_avoidance_by_lane_change_module/test_behavior_path_avoidance_by_lane_change_module.gtest.xml
1: Running main() from gmock_main.cc
1: [==========] Running 2 tests from 1 test suite.
1: [----------] Global test environment set-up.
1: [----------] 2 tests from PlanningModuleInterfaceTest
1: [ RUN      ] PlanningModuleInterfaceTest.NodeTestWithExceptionRoute
1: unknown file: Failure
1: C++ exception with description "failed to parse arguments: Couldn't parse params file: '--params-file /__w/autoware.universe/autoware.universe/install/behavior_path_planner/share/behavior_path_planner/config/lane_change/lane_change.param.yaml'. Error: Error opening YAML file, at ./src/parser.c:270, at ./src/rcl/arguments.c:406" thrown in the test body.
```

**Comment by satoshi-ota** [comment]
@zulfaqar-azmi-t4 https://github.com/autowarefoundation/autoware.universe/blob/3d40702b39fe12b64576393bae71e00753290eae/planning/behavior_path_avoidance_by_lane_change_module/test/test_behavior_path_planner_node_interface.cpp#L66-L67

**Review by satoshi-ota — APPROVED** [review]
@zulfaqar-azmi-t4 LGTM!

**Pull request merged**